rl4co/models/nn/ops.py

```diff
@@ -15,23 +16,24 @@ def forward(self, x):
 class Normalization(nn.Module):
     def __init__(self, embed_dim, normalization="batch"):
         super(Normalization, self).__init__()
-
-        normalizer_class = {"batch": nn.BatchNorm1d, "instance": nn.InstanceNorm1d}.get(
-            normalization, None
-        )
-
-        self.normalizer = normalizer_class(embed_dim, affine=True)
-
-    def init_parameters(self):
-        for name, param in self.named_parameters():
-            stdv = 1.0 / math.sqrt(param.size(-1))
-            param.data.uniform_(-stdv, stdv)
-
+        if normalization != 'layer':
+            normalizer_class = {"batch": nn.BatchNorm1d, "instance": nn.InstanceNorm1d}.get(
+                normalization, None
+            )
+    
+            self.normalizer = normalizer_class(embed_dim, affine=True)
+        else:
+            self.normalizer = 'layer'
+            
     def forward(self, x):
         if isinstance(self.normalizer, nn.BatchNorm1d):
             return self.normalizer(x.view(-1, x.size(-1))).view(*x.size())
         elif isinstance(self.normalizer, nn.InstanceNorm1d):
             return self.normalizer(x.permute(0, 2, 1)).permute(0, 2, 1)
+        elif self.normalizer == 'layer':
```

Comment:
This is tricky!! Basically the idea is exactly the LayerNorm. But the PyTorch LayerNorm requires pre-defining the shape of the tensors to norm, so in our case, we need to know the graph_size (the mean and var are computed wrt both graph_size and embed_dim). So i write my own normalisation. Also, the API for LayerNorm is different from other, e.g., it uses args like elementwise_affine rather than affine